### ISSUE TITLE
Enforce code coverage

### ISSUE THREAD

**Issue by Pwuts** [Significant-Gravitas/Auto-GPT#2455]
https://app.codecov.io/login/gh

**Comment by ntindle** [comment]
going to be workign on this after my pr is merged for shirts

**Comment by 7--** [comment]
How about fixing all the obvious bugs before wasting time covering code.

**Comment by Pwuts** [comment]
@7-- you're welcome to help. Grab a shovel and dig. 400 issues open, 250 PRs, those need filtering and deduplication before we can even prioritize them.

Also, where do you think the issues come from in the first place? It doesn't help that we don't have coverage and do have multiple "supported" ways to run Auto-GPT.

**Comment by Pwuts** [comment]
Duplicate of #2424

**Comment by Pwuts** [comment]
@jerrodcodecov it's more that we didn't have a lot of time to set it up so we haven't looked at many of the settings yet. :)

Where should we look to disable the project coverage check?

**Comment by jerrodcodecov** [comment]
Update from convo with @ntindle and @Pwuts 

1.) @vlad-ko and I are going to fork/PR a new change to your Codecov YAML based on the above

2.) @ntindle is going to change the Github Actions flow to upload a set of `unit` tests and a set of `integration` tests and then Codecov will use Carryforward Logic to Carryforward most recent integration test run when a fork doesn't have the appropriate API keys to run the integration tests

3.) @ntindle (or us) will make one more tweak to the Codecov YAML to ingest the flags correctly (we will put it in the current YAML change, commented out)
